### PR TITLE
Feature/write all lines in file system provider

### PR DESCRIPTION
**Pull request by MartinsRui**
Currently can not write tests for SftpFileSystemProvider since there is no interface for it.
Waiting on a pull request to be accepted on SSH.NET's github

### CODE REVIEW

**Comment by MartinsRui** [comment]
created the interface for ISftpClient locally

File: test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        public void Can_ListDirectory()
+        {
+            var files = _fileSystem.List(_filesPath);
+
+            A.CallTo(() => _fakeSftpClient.ListDirectory(_filesPath, null)).MustHaveHappened(Repeated.Exactly.Once);
+
+        }
+
+        [Fact]
+        public void Can_ListDirectoryWithSearchPattern()
```
Comment (pingvinen) [inline test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs]: The test does not prove that it returns only elements that match the given search pattern.

File: test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+        [Fact]
+        public void Can_CombinePaths()
+        {
+          Assert.Equal(_fileSystem.CombinePath(_assemblyPath, _localFilesFolder), Path.Combine(_assemblyPath, _localFilesFolder));
+        }
+
+        [Fact]
+        public void Can_OpenReadAFile()
```
Comment (pingvinen) [inline test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs:108]: The test does not check the return value.

File: test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        public void Can_ShowCWD()
+        {
+            var cwd = _fileSystem.Cwd();
+            Assert.NotNull(cwd);
+            Assert.NotEmpty(cwd);
+            Assert.Equal("not empty", cwd);
+        }
+
+      [Fact]
+        public void Can_ListDirectory()
```
Comment (pingvinen) [inline test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs:71]: The test does not check the return value.

File: test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            var fileContent = "my content";
+
+            var stream = _fileSystem.OpenReadFile(Path.Combine(_filesPath, _textFileName));
+
+            A.CallTo(() => _fakeSftpClient.OpenRead(Path.Combine(_filesPath, _textFileName))).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public void Can_LoadContentOfAFile()
```
Comment (pingvinen) [inline test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs]: The test does not check the return value.

File: test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+        [Fact]
+        public void Can_DetermineIfDirectoryExists()
+        {
+            _fileSystem.DirectoryExists(_filesPath);
+            A.CallTo(() => _fakeSftpClient.Exists(_filesPath)).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public void Can_CreateDirectory()
```
Comment (pingvinen) [inline test/unit/IO/FileSystem/SftpFileSystemProviderTest.cs]: The test does not check the return value.

**Review by pingvinen — CHANGES_REQUESTED** [review]

**Comment by MartinsRui** [comment]
@pingvinen one more time :)

**Review by pingvinen — DISMISSED** [review]

**Review by pingvinen — APPROVED** [review]

**Pull request merged**